code/MANAGER/MANAGER.py

```diff
@@ -1,38 +1,140 @@
+            pass
+        elif mode is PWR_Mode_e.TEMP_AMB:
+            log.info(f"Calibrating temperature ambient")
+            pass
+        elif mode is PWR_Mode_e.TEMP_BODY:
+            log.info(f"Calibrating temperature body")
+            pass
+
+
+    def executeMachineStatus(self) -> None:
+        '''Execute the state machine
+        Args:
+            - None
+        Returns:
+            - None
+        Raises:
+            - None
+        '''
+        check_calib = {PWR_Mode_e.VOLT_HS: False, PWR_Mode_e.VOLT_LS: False, PWR_Mode_e.CURR: False,
+                       PWR_Mode_e.TEMP_ANOD: False, PWR_Mode_e.TEMP_AMB: False, PWR_Mode_e.TEMP_BODY: False}
+        guided = False
+        term = TERM_c()
+
+        while self.__status is not TERM_Option_e.EXIT:
+            #Guided mode option
+            if guided is True:
+                option_guided = [None, TERM_Option_e.FLASH_ORIG, TERM_Option_e.CONF_DEV, TERM_Option_e.CALIB, TERM_Option_e.FLASH_CALIB, TERM_Option_e.EXIT]
+                self.__status = option_guided[option_guided.index(self.__status) + 1]
+            else:
+                self.__status = term.queryOptions()
+
+            #Flash original program
+            if self.__status is TERM_Option_e.FLASH_ORIG:
+                log.info(f"Flashing original program")
+                pass
 
-    def _calibStatus(self):
-        pass
+            #Configure device
+            elif self.__status is TERM_Option_e.CONF_DEV:
+                log.info(f"Configuring device")
+                self.__epc_config = term.queryEPCConf()
 
+            #Calibrate device
+            elif self.__status is TERM_Option_e.CALIB:
+                log.info(f"Calibrating device")
+                if self.__epc_config is None:
+                    log.error("EPC configuration not set")
+                else:
+                    if guided is True:
+                        options_calib_mode = [PWR_Mode_e.VOLT_HS, PWR_Mode_e.VOLT_LS, PWR_Mode_e.CURR, PWR_Mode_e.TEMP_ANOD, PWR_Mode_e.TEMP_AMB, PWR_Mode_e.TEMP_BODY]
+                        for calib_mode in options_calib_mode:
+                            self._calibStatus(calib_mode)
+                            check_calib[calib_mode] = True
+                    else:
+                        calib_mode = self.__status = term.queryCalibMode()
+                        self._calibStatus(calib_mode)
+                        check_calib[calib_mode] = True
 
-    def executeMachineStatus(self):
-        pass
\ No newline at end of file
+            #Flash with calibration data
+            elif self.__status is TERM_Option_e.FLASH_CALIB:
+                log.info(f"Flashing with calibration data")
+                if self.__epc_config is None:
+                    log.error("EPC configuration not set")
+                else:
+                    if False in check_calib.values():
+                        log.error("Not all calibration data are set")

```

Comment:
If device not calibrate, execture calib status

Reply:
Done.